### PR TITLE
Add working version of kibana

### PR DESCRIPTION
**Pull request by Scriptkiddi**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
```
Comment (SlothOfAnarchy) [inline .travis.yml]: are we using this?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Ansible Role: Kibana
+
+An Ansible Role that installs Kibana on Ubuntu.
+
+## Requirements
+
+None.
```
Comment (SlothOfAnarchy) [inline README.md]: Will this work on all distributions?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+For variables please see the [kibana documentation](https://www.elastic.co/guide/en/kibana/current/settings.html) 
+
+
+## Dependencies
+None.
+
+## Example Playbook
+
+    - hosts: kibana
```
Comment (SlothOfAnarchy) [inline README.md]: Please quote this code (see our ansible skeleton).

File: handlers/main.yml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+---
+- name: restart kibana
```
Comment (SlothOfAnarchy) [inline handlers/main.yml]: Capitalize

File: handlers/main.yml [hunk 5]
```diff
@@ -0,0 +1,3 @@
...
+  service: name=kibana state=restarted
```
Comment (SlothOfAnarchy) [inline handlers/main.yml]: Please don't write all yml in one line. Change this everywhere.

File: meta/main.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+---
+dependencies: []
+
+galaxy_info:
+  author: Fritz Otlinghaus
+  description: Kibana instance.
+  company: "stuvus"
+  license: "license (BSD, MIT)"
```
Comment (SlothOfAnarchy) [inline meta/main.yml]: I thought it was BY-NC?
Please populate this file exactly like in our skeleton.

File: tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,24 @@
+---
+- include: setup-RedHat.yml
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: Where is this include or the support information in your meta file?

File: tasks/setup-Debian.yml [hunk 8]
```diff
@@ -0,0 +1,14 @@
+---
+- name: Ensure dependency is installed (Ubuntu).
+  apt: name=apt-transport-https state=present
```
Comment (SlothOfAnarchy) [inline tasks/setup-Debian.yml]: `state: present` is the default.

File: tasks/setup-Debian.yml [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+
+- name: Add Elasticsearch apt key.
+  apt_key:
+    url: https://packages.elastic.co/GPG-KEY-elasticsearch
+    state: present
```
Comment (SlothOfAnarchy) [inline tasks/setup-Debian.yml]: `state: present` is the default.

File: tasks/setup-Debian.yml [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+
+- name: Add Kibana repository.
+  apt_repository:
+    repo: 'deb https://packages.elastic.co/kibana/6.x/debian stable main'
+    state: present
```
Comment (SlothOfAnarchy) [inline tasks/setup-Debian.yml]: `state: present` is the default.

File: tasks/main.yml [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+  when: ansible_os_family == 'RedHat'
+
+- include: setup-Debian.yml
+  when: ansible_os_family == 'Debian'
+
+- name: Install Kibana.
+  package: name=kibana state=present
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: `state: present` is the default.

File: templates/kibana.yml.j2 [hunk 12]
```diff
@@ -0,0 +1,174 @@
...
+{% endif %}
+
+# The Kibana server's name.  This is used for display purposes.
+#server.name: "your-hostname"
+{% if kibiana_server_name is defined %}
+server.name: "{{ kibiana_server_name }}"
+{% endif %}
+
+# The URL of the Elasticsearch instance to use for all your queries.
+elasticsearch.url: {{ kibana_elasticsearch_url | mandatory }}
```
Comment (SlothOfAnarchy) [inline templates/kibana.yml.j2]: Please mention mandatory and the most-used variables in your readme.

File: README.md [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+      roles:
+        - stuvusIT.kibana
+
+## License
+
+This work is licensed under a [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/).
```
Comment (SlothOfAnarchy) [inline README.md]: I can't find the LICENSE file.

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

**Review by dasJ — DISMISSED** [review]
Requesting changes until you fix the issues mentioned by @SlothOfAnarchy and I get to review it

File: .travis.yml (hunk 1 above)
Comment (Scriptkiddi) [reply .travis.yml]: no we dont its deleted

File: README.md (hunk 2 above)
Comment (Scriptkiddi) [reply README.md]: yes

File: handlers/main.yml (hunk 4 above)
Comment (Scriptkiddi) [reply handlers/main.yml]: fixed

File: handlers/main.yml (hunk 5 above)
Comment (Scriptkiddi) [reply handlers/main.yml]: fixed

File: tasks/main.yml (hunk 7 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: line removed

File: tasks/main.yml (hunk 11 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: fixed

File: tasks/setup-Debian.yml (hunk 8 above)
Comment (Scriptkiddi) [reply tasks/setup-Debian.yml]: fixed

File: tasks/setup-Debian.yml (hunk 10 above)
Comment (Scriptkiddi) [reply tasks/setup-Debian.yml]: fixed

File: README.md (hunk 3 above)
Comment (SlothOfAnarchy) [reply README.md]: again, please use block quotes. Just orientate yourself on out skeleton

File: README.md (hunk 13 above)
Comment (SlothOfAnarchy) [reply README.md]: Still can't find the license file.

File: README.md [hunk 14]
```diff
@@ -1,24 +1,38 @@
...
 
+### Mandatory variables
+Some variables are only possible to configure together they are also listed below
+| Name                                    |         Required         | Default | Description                                                                                                                                                            |    |
+|:----------------------------------------|:------------------------:|:--------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:---|
+| `kibana_elasticsearch_url `             |    :heavy_check_mark:    |         | The URL of the Elasticsearch instance to use for all your queries                                                                                                      |    |
+| `kibiana_elasticsearch_password`        | :heavy_multiplication_x: |         | If your Elasticsearch is protected with basic authentication, these settings provide the username and password. Required when `kibiana_elasticsearch_username` is set. |    |
```
Comment (SlothOfAnarchy) [inline README.md]: kib**i**ana? 
Same below.

File: README.md [hunk 15]
```diff
@@ -1,24 +1,38 @@
...
+| `kibiana_server_ssl_certificate`        | :heavy_multiplication_x: |         | Required if `kibiana_server_ssl_enabled` is set. SSL certificate for the Kibana server                                                                                 |    |
+| `kibiana_server_ssl_key`                | :heavy_multiplication_x: |         | Required if `kibiana_server_ssl_enabled` is set. SSL key for the Kibana server                                                                                         |    |
+| `kibiana_elasticsearch_ssl_certificate` | :heavy_multiplication_x: |         | These files validate that your Elasticsearch backend uses the same key files.                                                                                          |    |
```
Comment (SlothOfAnarchy) [inline README.md]: certificate files instead of key files, I think

File: README.md [hunk 16]
```diff
@@ -1,24 +1,38 @@
...
+| `kibiana_elasticsearch_ssl_key`         | :heavy_multiplication_x: |         | Required if `kibiana_elasticsearch_ssl_certificate` is set. These files validate that your Elasticsearch backend uses the same key files.                              |    |
 
-## Dependencies
-None.
 
 ## Example Playbook
 
+Including an example of how to use your role (for instance, with variables passed in as parameters) is always nice for users too:
```
Comment (SlothOfAnarchy) [inline README.md]: Change or delete this paragraph.

File: README.md [hunk 17]
```diff
@@ -27,4 +41,4 @@ This work is licensed under a [Creative Commons Attribution-ShareAlike 4.0 Inter
 
 ## Author Information
 
-- [Fritz Otlinghaus (Scriptkiddi)](https://github.com/scriptkiddi) _fritz.otlinghaus@stuvus.uni-stuttgart.de_
+- [Fritz Otlinghaus (Scriptkidid)](https://github.com/scriptkiddi) _fritz.otlinghaus@stuvus.uni-stuttgart.de_
```
Comment (SlothOfAnarchy) [inline README.md]: Did you change your nickname?

File: README.md [hunk 18]
```diff
@@ -1,24 +1,38 @@
...
+| `kibana_elasticsearch_url `             |    :heavy_check_mark:    |         | The URL of the Elasticsearch instance to use for all your queries                                                                                                      |    |
+| `kibiana_elasticsearch_password`        | :heavy_multiplication_x: |         | If your Elasticsearch is protected with basic authentication, these settings provide the username and password. Required when `kibiana_elasticsearch_username` is set. |    |
+| `kibiana_server_ssl_certificate`        | :heavy_multiplication_x: |         | Required if `kibiana_server_ssl_enabled` is set. SSL certificate for the Kibana server                                                                                 |    |
+| `kibiana_server_ssl_key`                | :heavy_multiplication_x: |         | Required if `kibiana_server_ssl_enabled` is set. SSL key for the Kibana server                                                                                         |    |
```
Comment (SlothOfAnarchy) [inline README.md]: These vars are marked as mandatory when used

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

File: README.md [hunk 19]
```diff
@@ -36,9 +36,51 @@ Including an example of how to use your role (for instance, with variables passe
 
 ## License
 
-This work is licensed under a [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/).
+This work is licensed under a [Creative Commons Attkibanaribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/).
```
Comment (SlothOfAnarchy) [inline README.md]: ...

**Review by SlothOfAnarchy — DISMISSED** [review]

File: README.md [hunk 20]
```diff
@@ -0,0 +1,87 @@
+# Kibana 6
+
+This role sets up a running kibana instance and configures it.
+
+
```
Comment (ghost) [inline README.md]: Remove this empty line

File: README.md [hunk 21]
```diff
@@ -0,0 +1,87 @@
...
+## Requirements
+
+None
+
+
```
Comment (ghost) [inline README.md]: Remove this empty line

File: README.md [hunk 22]
```diff
@@ -0,0 +1,87 @@
...
+## Role Variables
+
+For variables please see the [kibana documentation](https://www.elastic.co/guide/en/kibana/current/settings.html)
```
Comment (ghost) [inline README.md]: Capitalize Kibana, also add a comma after `variables`

File: README.md [hunk 23]
```diff
@@ -0,0 +1,87 @@
...
+
+### Mandatory variables
+Some variables are only possible to configure together they are also listed below
+
+| Name                                    |         Required         | Default | Description                                                                                                                                                            |    |
```
Comment (ghost) [inline README.md]: Additional columns

File: README.md [hunk 24]
```diff
@@ -0,0 +1,87 @@
...
+|:----------------------------------------|:------------------------:|:--------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:---|
+| `kibana_elasticsearch_url `             |    :heavy_check_mark:    |         | The URL of the Elasticsearch instance to use for all your queries                                                                                                      |    |
```
Comment (ghost) [inline README.md]: Are these really my queries? What if somebody else logs on and performs queries?

File: README.md [hunk 25]
```diff
@@ -0,0 +1,87 @@
...
+| `kibana_elasticsearch_password`        | :heavy_multiplication_x: |         | If your Elasticsearch is protected with basic authentication, these settings provide the username and password. Required when `kibana_elasticsearch_username` is set. |    |
+| `kibana_server_ssl_certificate`        | :heavy_multiplication_x: |         | Required if `kibana_server_ssl_enabled` is set. SSL certificate for the Kibana server                                                                                 |    |
+| `kibana_server_ssl_key`                | :heavy_multiplication_x: |         | Required if `kibana_server_ssl_enabled` is set. SSL key for the Kibana server                                                                                         |    |
+| `kibana_elasticsearch_ssl_certificate` | :heavy_multiplication_x: |         | These files validate that your Elasticsearch backend uses the same key files.                                                                                          |    |
```
Comment (ghost) [inline README.md]: Are those multiple files?

File: README.md [hunk 26]
```diff
@@ -0,0 +1,87 @@
...
+| `kibana_elasticsearch_ssl_key`         | :heavy_multiplication_x: |         | Required if `kibana_elasticsearch_ssl_certificate` is set. These files validate that your Elasticsearch backend uses the same key files.                              |    |
```
Comment (ghost) [inline README.md]: Again

File: README.md [hunk 27]
```diff
@@ -0,0 +1,87 @@
...
+
+
```
Comment (ghost) [inline README.md]: Remove empty line

File: README.md [hunk 28]
```diff
@@ -0,0 +1,87 @@
...
+## Example Playbook
+
+Including an example of how to use your role (for instance, with variables passed in as parameters) is always nice for users too:
```
Comment (ghost) [inline README.md]: What is this line?

File: README.md [hunk 29]
```diff
@@ -0,0 +1,87 @@
...
+
+```yml
```
Comment (ghost) [inline README.md]: Don't you have a more sophisticated example?

File: README.md [hunk 30]
```diff
@@ -0,0 +1,87 @@
...
+    - hosts: kibana
+      roles:
+        - stuvusIT.kibana
+```
+
+## License
+
+This work is licensed under a [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/).
+
+
```
Comment (ghost) [inline README.md]: Empty line

File: README.md [hunk 31]
```diff
@@ -0,0 +1,87 @@
...
+## Author Information
+
+- [Fritz Otlinghaus (Scriptkidid)](https://github.com/scriptkiddi) _fritz.otlinghaus@stuvus.uni-stuttgart.de_
+# Kibana 6
+
+This role sets up a running kibana instance and configures it.
+
+
```
Comment (ghost) [inline README.md]: Empty line

File: README.md [hunk 32]
```diff
@@ -0,0 +1,87 @@
...
+## License
+
+This work is licensed under a [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/).
+
+
+## Author Information
+
+- [Fritz Otlinghaus (Scriptkidid)](https://github.com/scriptkiddi) _fritz.otlinghaus@stuvus.uni-stuttgart.de_
+# Kibana 6
```
Comment (ghost) [inline README.md]: Again?

File: handlers/main.yml [hunk 33]
```diff
@@ -0,0 +1,5 @@
+---
+- name: Restart kibana
```
Comment (ghost) [inline handlers/main.yml]: *Kibana

File: tasks/main.yml [hunk 34]
```diff
@@ -0,0 +1,35 @@
+---
+- name: Ensure dependency is installed (Ubuntu).
```
Comment (ghost) [inline tasks/main.yml]: No dot

File: tasks/main.yml [hunk 35]
```diff
@@ -0,0 +1,35 @@
...
+  apt: 
+    name: apt-transport-https
+  when: ansible_os_family == 'Debian'
+
+- name: Add Elasticsearch apt key.
```
Comment (ghost) [inline tasks/main.yml]: No Dot

File: tasks/main.yml [hunk 36]
```diff
@@ -0,0 +1,35 @@
...
+  apt_key:
+    url: https://packages.elastic.co/GPG-KEY-elasticsearch
+  when: ansible_os_family == 'Debian'
+
+- name: Add Kibana repository.
```
Comment (ghost) [inline tasks/main.yml]: No Dot

File: tasks/main.yml [hunk 37]
```diff
@@ -0,0 +1,35 @@
...
+  apt_repository:
+    repo: 'deb https://packages.elastic.co/kibana/6.x/debian stable main'
+    update_cache: yes
+  when: ansible_os_family == 'Debian'
+
+- name: Install Kibana.
```
Comment (ghost) [inline tasks/main.yml]: No Dot

File: tasks/main.yml [hunk 38]
```diff
@@ -0,0 +1,35 @@
...
+  package: 
+    name: kibana
+
+- name: Ensure Kibana is started and enabled at boot.
```
Comment (ghost) [inline tasks/main.yml]: No Dot

File: tasks/main.yml [hunk 39]
```diff
@@ -0,0 +1,35 @@
...
+  service:
+    name: kibana
+    state: started
+    enabled: yes
+
+- name: Copy Kibana configuration.
```
Comment (ghost) [inline tasks/main.yml]: No Dot

File: tasks/main.yml [hunk 40]
```diff
@@ -0,0 +1,35 @@
...
+  template:
+    src: kibana.yml.j2
+    dest: "/opt/kibana/config/kibana.yml"
+    owner: root
+    group: root
+    mode: 0644
+  notify: restart kibana
```
Comment (ghost) [inline tasks/main.yml]: Capitalize both words

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by SlothOfAnarchy** [comment]
@Scriptkiddi what's the state on this?

File: README.md (hunk 20 above)
Comment (Scriptkiddi) [reply README.md]: done

File: README.md (hunk 21 above)
Comment (Scriptkiddi) [reply README.md]: done

File: README.md (hunk 22 above)
Comment (Scriptkiddi) [reply README.md]: done

File: README.md (hunk 24 above)
Comment (Scriptkiddi) [reply README.md]: Same comment as in the Kibana docs

File: README.md (hunk 23 above)
Comment (Scriptkiddi) [reply README.md]: Yes old layout wont fix

File: README.md (hunk 28 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 27 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 25 above)
Comment (Scriptkiddi) [reply README.md]: As in the kibana config yes

File: README.md (hunk 30 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 31 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 32 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 29 above)
Comment (Scriptkiddi) [reply README.md]: no i dont

File: tasks/main.yml (hunk 35 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: why not

File: tasks/main.yml (hunk 36 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: why not

File: tasks/main.yml (hunk 35 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: ignore pls and fixed

File: tasks/main.yml (hunk 34 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: fixed

File: tasks/main.yml (hunk 36 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: fixed

File: tasks/main.yml (hunk 37 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: fixed

File: tasks/main.yml (hunk 39 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: fixed

File: tasks/main.yml (hunk 40 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: fixed

**Comment by Scriptkiddi** [comment]
@dasJ pls rereview

**Review by ghost — APPROVED** [review]

**Pull request merged**